### DISCUSSION TITLE
Consider deprecating alternate Store API endpoints

### DISCUSSION THREAD

**Discussion by senadir** [woocommerce/woocommerce#45042] [WooCommerce, Blocks, and Site Editing]
Store API ships with a couple of REST-complaint endpoints for managing cart items and coupons.

```
		'/wc/store/v1/cart/coupons': [ 'GET', 'POST', 'DELETE' ],
		'/wc/store/v1/cart/coupons/(?P<code>[\\w-]+)': [ 'GET', 'DELETE' ],
		'/wc/store/v1/cart/items': [ 'GET', 'POST', 'DELETE' ],
		'/wc/store/v1/cart/items/(?P<key>[\\w-]{32})': [
			'GET',
			'POST',
			'PUT',
			'PATCH',
			'DELETE',
		],
```

Those endpoints would either return the item created/updated only, or 204 code when deleting an item. Calling `DELETE cart/items` or `DELETE cart/coupons` would empty the cart or would delete all coupons.

We don't use those endpoints at all in Cart/Checkout because they don't return an updated cart and no cart totals, meaning you need to manage cart state manually.

The cart is updated regardless on server, so they provide little performance benefit over other endpoints.

### Option 1: Deprecate them

For now, they still exist but get no attention at all, to limit the risk of people using them, we should consider deprecating and deleting them overtime.

Developers should use the `cart/add-item` `cart/update-item` `cart/remove-item` `cart/apply-coupon` `cart/remove-coupon`

### Option 2: Let them return Cart
To make them useful again, we can consider having them return Cart and deprecate the other endpoints in favor of those.

cc @mikejolley

**Comment by mikejolley** [comment]
These endpoints return item data, rather than cart as a whole. Thats why we don't consume them in blocks, but they still have uses. For example, deleting cart items is something we've used in e2e tests.

I guess look at the last part of the request. `/cart/coupons` will return coupon data, whereas anything ending with `/cart` returns cart data. I realise there are some exceptions to this in the current codebase for the more transactional routes.
